core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/FastPassMonitor.scala

```diff
@@ -0,0 +1,80 @@
+package org.broadinstitute.dsde.rawls.monitor
+import akka.actor.{Actor, Props}
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.dataaccess.slick.DataAccess
+import org.broadinstitute.dsde.rawls.model.{FastPassGrant, GcpResourceTypes, MemberTypes, Workspace}
+import org.broadinstitute.dsde.rawls.monitor.FastPassMonitor.DeleteExpiredGrants
+import org.broadinstitute.dsde.workbench.google.GoogleIamDAO.MemberType
+import org.broadinstitute.dsde.workbench.google.{GoogleIamDAO, GoogleStorageDAO}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+
+import scala.language.postfixOps
+
+object FastPassMonitor {
+  sealed trait FastPassMonitorMessage
+  case object DeleteExpiredGrants extends FastPassMonitorMessage
+  def props(dataSource: SlickDataSource, googleIamDao: GoogleIamDAO, googleStorageDao: GoogleStorageDAO): Props = Props(
+    new FastPassMonitor(dataSource, googleIamDao, googleStorageDao)
+  )
+}
+
+class FastPassMonitor private (dataSource: SlickDataSource,
+                               googleIamDao: GoogleIamDAO,
+                               googleStorageDao: GoogleStorageDAO
+) extends Actor {
+  import context.dispatcher
+  override def receive: Receive = { case DeleteExpiredGrants =>
+    deleteExpiredGrants()
+  }
+
+  private def deleteExpiredGrants(): Unit =
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.fastPassGrantQuery.findExpiredFastPassGrants().map { grants =>
+        grants.to(LazyList).groupBy(_.workspaceId).foreach { case (workspaceId, grants) =>
+          dataAccess.workspaceQuery.findById(workspaceId).map { maybeWorkspace =>
+            val workspace =
+              maybeWorkspace.getOrElse(throw new RuntimeException(s"Could not find workspace $workspaceId"))
+            grants.groupBy(_.accountEmail).foreach { case (_, grants) =>
+              removeGrantsForAccountEmailInWorkspace(dataAccess, workspace, grants)
+            }
+          }
+        }
+      }
+    }
+
+  private def removeGrantsForAccountEmailInWorkspace(dataAccess: DataAccess,
+                                                     workspace: Workspace,
+                                                     grants: LazyList[FastPassGrant]
+  ): Unit = {
+    val organizationRoles = grants.map(_.organizationRole).toSet
+    grants foreach { grant =>
+      val memberType: MemberType = matchGrantMemberType(grant)
+      grant.resourceType match {
+        case GcpResourceTypes.Project =>
+          googleIamDao.removeIamRoles(GoogleProject(workspace.googleProjectId.value),
+                                      WorkbenchEmail(grant.accountEmail.value),
+                                      memberType,
+                                      organizationRoles
```

Comment:
The `organizationRole`s for projects and buckets are different, so you'll have to group the `organizationRoles` by resource type